### PR TITLE
[changelog] add possibility to overwrite previous tag Resolves #382

### LINKED ISSUE #382: [changelog] Tag/Version range for changelog
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**
I would like to define a range of tags/versions to find all commits in between to generate the changelog.

**Describe alternatives you've considered**

**Additional context**
We are creating tags for every push to master, but are creating gitlab releases after we have a couple of features/fixes in the master. So our release changelog would consist of multiple tags.

### PR DESCRIPTION
**Pull request by andreabettich**
Resolves #382 

### Checklist
- [x] [Review Contribution Guidelines](https://github.com/jreleaser/jreleaser/blob/master/CONTRIBUTING.md).
- [x] Make sure all contributed code can be distributed under the terms of the 
      [Apache License 2.0](https://github.com/jreleaser/jreleaser/blob/master/LICENSE), e.g. the code was written by 
      you or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Update [documentation](https://github.com/jreleaser/jreleaser.github.io) when applicable.

### CODE REVIEW

File: sdks/git-sdk/src/main/java/org/jreleaser/sdk/git/ChangelogGenerator.java [hunk 1]
```diff
@@ -238,10 +239,19 @@ private static Comparable version(JReleaserContext context, Ref tag, Pattern ver
...
         Comparable currentVersion = version(context, tag.get(), versionPattern);
 
-        Optional<Ref> previousTag = tags.stream()
-            .filter(ref -> extractTagName(ref).matches(tagPattern))
-            .filter(ref -> lessThan(version(context, ref, versionPattern), currentVersion))
-            .findFirst();
+        String previousTagName = gitService.getConfiguredPreviousTagName();
+        Optional<Ref> previousTag;
+
+        if(StringUtils.isNotBlank(previousTagName)){
```
Comment (aalmiray) [inline sdks/git-sdk/src/main/java/org/jreleaser/sdk/git/ChangelogGenerator.java:245]: Please preserve white space as every other `if` statement does

File: plugins/jreleaser-gradle-plugin/src/main/groovy/org/jreleaser/gradle/plugin/internal/dsl/AbstractGitService.groovy [hunk 2]
```diff
@@ -56,6 +56,7 @@ abstract class AbstractGitService implements GitService {
     final Property<String> username
     final Property<String> token
     final Property<String> tagName
+    final Property<String> previousTagName
```
Comment (aalmiray) [inline plugins/jreleaser-gradle-plugin/src/main/groovy/org/jreleaser/gradle/plugin/internal/dsl/AbstractGitService.groovy:59]: Please add a `Property<String> getPreviousTagName()` to the `GitService.groovy` DSL interface.

**Comment by aalmiray** [comment]
Thank you @andreabettich. You found most places to make the required changes, only missing updates for the Maven plugin. The following types must be updated as well:

https://github.com/jreleaser/jreleaser/blob/ba1ed9b50d2921789b9b95cdb4b583321d1a5504/plugins/jreleaser-maven-plugin/src/main/java/org/jreleaser/maven/plugin/GitService.java
https://github.com/jreleaser/jreleaser/blob/ba1ed9b50d2921789b9b95cdb4b583321d1a5504/plugins/jreleaser-maven-plugin/src/main/java/org/jreleaser/maven/plugin/JReleaserAutoConfigReleaseMojo.java
https://github.com/jreleaser/jreleaser/blob/ba1ed9b50d2921789b9b95cdb4b583321d1a5504/plugins/jreleaser-maven-plugin/src/main/java/org/jreleaser/maven/plugin/internal/JReleaserModelConverter.java

**Comment by andreabettich** [comment]
I fixed the formatting and added the missing parts. Sorry for already amending and force-pushing onto the branch. makes it less transparent what I changed.

**Comment by aalmiray** [comment]
Thank you. Force push is OK. I'll do one full review again and then we'll be good to go. 

If you feel like doing so, you could also send a PR to jreleaser/jreleaser.github.io for updating configuration snippets. If not that's OK, I can do that later 😄

**Comment by andreabettich** [comment]
Added some documentation https://github.com/jreleaser/jreleaser.github.io/pull/16

**Comment by aalmiray** [comment]
Thank you for your contribution! 🎉 
You don't have to wait til `v0.8.0` is released to test out this feature. You can grab the latest `early-access` release and invoke the CLI directly. Alternatively the jreleaser/jreleaser-slim:early-access Docket image should work with GitLab CI as well

https://hub.docker.com/r/jreleaser/jreleaser-slim/tags

Just have tom wait til the current early-access build finishes to grab the binaries.

**Comment by andreabettich** [comment]
awesome, thanks a lot!

**Pull request merged**